Dockerfile

```diff
@@ -1,5 +1,9 @@
 FROM python:3-slim
 
+ENV SHELL_INTERACTION=false
+ENV OVERWRITE_OS_NAME=""
+ENV OVERWRITE_SHELL_NAME=""
+
```

Comment:
```suggestion
ARG OVERWRITE_OS_NAME=default
ARG OVERWRITE_SHELL_NAME=default

ENV SHELL_INTERACTION=false
ENV OVERWRITE_OS_NAME=${OVERWRITE_OS_NAME}
ENV OVERWRITE_SHELL_NAME=${OVERWRITE_SHELL_NAME}

```

Reply:
I believe you misunderstand Docker's ARG:
These are env variables that only exist during the building process of the container image.
Once the image exists, these are gone forever.
Please refer the [dockerfile arg documentation](https://docs.docker.com/reference/dockerfile/#arg) for a better, more complete explanation.

ENV sets environment variables that remain, and can be used set as arguments with --env as desired when starting the container

Reply:
You are right, please ignore this comment.